lib/utils.sh

```diff
@@ -295,52 +295,68 @@ is_service_enabled() {
 is_kernel_option_enabled() {
     local KERNEL_OPTION="$1"
     local MODULE_NAME=""
+    local FILTER=""
     local RESULT=""
+    local IS_MONOLITHIC_KERNEL=1
+    local DEF_MODULE=""
 
     if [ $# -ge 2 ]; then
         MODULE_NAME="$2"
     fi
 
-    if $SUDO_CMD [ -r "/proc/config.gz" ]; then
-        RESULT=$($SUDO_CMD zgrep "^$KERNEL_OPTION=" /proc/config.gz) || :
-    elif $SUDO_CMD [ -r "/boot/config-$(uname -r)" ]; then
-        RESULT=$($SUDO_CMD grep "^$KERNEL_OPTION=" "/boot/config-$(uname -r)") || :
-    else
-        debug "No information about kernel found, you're probably in a container"
-        FNRET=127
-        return
+    if [ $# -ge 3 ]; then
+        FILTER="$3"
     fi
 
-    ANSWER=$(cut -d = -f 2 <<<"$RESULT")
-    if [ "x$ANSWER" = "xy" ]; then
-        debug "Kernel option $KERNEL_OPTION enabled"
-        FNRET=0
-    elif [ "x$ANSWER" = "xn" ]; then
-        debug "Kernel option $KERNEL_OPTION disabled"
-        FNRET=1
-    else
-        debug "Kernel option $KERNEL_OPTION not found"
-        FNRET=2 # Not found
+    if $SUDO_CMD lsmod >/dev/null 2>&1; then
+        IS_MONOLITHIC_KERNEL=0
     fi
 
-    if $SUDO_CMD [ "$FNRET" -ne 0 ] && [ -n "$MODULE_NAME" ] && [ -d "/lib/modules/$(uname -r)" ]; then
```

Comment:
Why did you remove this part ? I think it's still relevant (in both cases)

Reply:
Regarding CIS recommendations if you override module (with install /bin/(true|false)) and lsmod return nothing, this is considered as secure, with your version we add kernel option detection for monolithic kernel.

If you consider we should detect if modules file exists, check if blacklist or override, this is redundant if `lsmod/modprobe` work but maybe you have case (example WSL) where a monolithic kernel don't have a /proc/modules and can use dynamic modules so I propose to restore check only if `lsmod/modprobe` doesn't work, would that be suitable ? (if yes check my last commit)

PS: regarding modprobe.d manpage (https://linux.die.net/man/5/modprobe.d) if I restore the check I should change blacklist/install `find` command to only check `.conf` file (`All files underneath the /etc/modprobe.d directory which end with the .conf extension specify those options as required`)

Reply:
> If you consider we should detect if modules file exists, check if blacklist or override, this is redundant if lsmod/modprobe work but maybe you have case (example WSL) where a monolithic kernel don't have a /proc/modules and can use dynamic modules so I propose to restore check only if lsmod/modprobe doesn't work, would that be suitable ? (if yes check my last commit)
Yes that was what I meant
Ok for the  èfind`